sys/net/application_layer/gcoap/fileserver.c

```diff
@@ -644,4 +695,16 @@ ssize_t gcoap_fileserver_handler(coap_pkt_t *pdu, uint8_t *buf, size_t len,
     return gcoap_response(pdu, buf, len, errorcode);
 }
 
+#ifdef MODULE_GCOAP_FILESERVER_CALLBACK
```

Comment:
maybe move the `#ifdef` inside the function

Reply:
Huh, why?
I'd actually prefer this to be a compile-time error when the function is called without the module enable instead of it silently turning into a no-op.

Reply:
I find it more readable if a function call does not have to be guarded with a module conditional.
If you prefer it to be a compile time error, I will not be nitpicky.